pkg/apis/clusteroperator/types.go

```diff
@@ -412,10 +416,6 @@ type ClusterDeploymentStatus struct {
 	// ClusterVersionRef references the resolved clusterversion the cluster should be running.
 	// +optional
 	ClusterVersionRef *corev1.ObjectReference
-
```

Comment:
Should Conditions be of type []ClusterDeploymentCondition and only include ClusterDeployment-specific type of conditions there?

Reply:
In addition to duplicating the type structs, I think we'd need to duplicate a few functions like https://github.com/openshift/cluster-operator/blob/master/pkg/controller/controller_utils.go#L155. What do you think?

Reply:
agree

Reply:
Agree as in lets fork the code or should we leave it as is? Assuming it was a +1 for forking.